dynawo/sources/Modeler/Common/DYNSubModel.cpp

```diff
@@ -1186,7 +1186,9 @@ SubModel::addMessage(const string& message) {
 
 void
 SubModel::addEvent(const string& modelName, const MessageTimeline& messageTimeline) {
```

Comment:
We still build the MessageTimeline instance here. 
What I was proposing to try is to create 2 versions of addEvent:
SubModel::addEvent(const string& modelName, const string& message)
SubModel::addEvent(const string& modelName, KeyTimeline_t key, ...)
the add Event body will then be something like:
if (hasTimeline()) {
messageTimeline messageTimeline(...)
timeline_->addEvent(getCurrentTime(), modelName, messageTimeline.str(), messageTimeline.priority())
}
and then to call those in the models instead of the current version of addEvent

Reply:
I think I get your point... Tell me if what you ask is to get rid of the class called MessageTimeline ? Or maybe you don't want that a Timeline object/reference/pointer is passed as an argument to addEvent() (?) Until now I understood that you  did not want to create a MessageTimeline before/when calling addEvent. Not calling addEvent resolved that last point.

Reply:
I would like to try a solution as follows:

On model side
network-> addEvent(... ) (no test of hasTimeline, way better for readability and maintenability)

On add Event side :
if (hasTimeline()) {
messageTimeline messageTimeline(...)
timeline_->addEvent(getCurrentTime(), modelName, messageTimeline.str(), messageTimeline.priority())
}

Reply:
On Model side, addEvent will be addEvent( string, string) or addEvent( string, key_t) ...
And I will pass the arguments to the ctor of messasgeTimeline only in cas we need to log an event. That's really easy to do, why didn't you tell ?

Reply:
no comment

Reply:
done, I even added a unitary test